New guide for IAM logs (#5314)
ovh/docs: pages/manage_and_operate/iam/iam-logs-forwarding/guide.en-gb.md

---
title: OVHcloud Account logs Forwarding to LDP
excerpt: "Forwarding OVHcloud Account logs to Logs Data Platform"
updated: 2023-10-23
---

 

> ## Objectif

Reviewer: here and L14 are in french

**Logs forwarding**: feature integrated in an OVHcloud product to ingest logs from its services to a data stream of an LDP account in the same OVHcloud account. The feature has to be enabled by the customer per service.

**Logs forwarding Subscription**: when enabling the logs forwarding for a given OVHcloud service to a given LDP data stream, a subscription is created and attached to the data stream for further management by the customer.

**OVH Account audit logs**: Log entries for sensible activities happening within your OVH Account customer main need to collect and use for compliance purposes and other business policy enforcement.

**OVH Account activity logs**: Log entries for all API calls done through OVHCloud API including those triggered when using OVHcloud manager.

**OVH Account policy logs**: Log entries for OVH services access acceptance or denied access due to of a security policy evaluation.

## Instruction

### OVHcloud Account logs type

OVHcloud account propose 3 level of logs
-	**Audit Logs**: provide a security-relevant, chronological set of records documenting the sequence of actions in your OVHCLOUD Account. (ie. logins, password change, …)

> -	**Activity Logs**: provide all of records of actions in your OVHCLOUD Account. (ie. API calls and actions done on the control panel

Reviewer: missing closing parenthesis

-	**Access policy logs**: provide all of records of access evaluation in your OVHCLOUD Account. (ie. Actions authorized or unauthorized through IAM policies)

### Enable logs forwarding

> Currently, you can only enable the forwarding of the OVHcloud account logs via API. You will have to define the target Stream of one of your LDP account you aim these logs to be forwarded to. The enablement of the forwarding will create a subscription for this stream id.

Reviewer: ```suggestion
Currently, you can only enable the forwarding of the OVHcloud account logs via API. You will have to target a Stream of one of your LDP account. The logs will be forwarded to that Stream. Enabling the forwarding will create a subscription for this stream id.
```

> Note that the forwarding enablement is free of charge, but you will be charged for the usage of Logs Data Platform service as per standard price plan. For LDP pricing refer to [Logs Data Platform product page](https://www.ovhcloud.com/en-ie/logs-data-platform/).

Reviewer: Note that enabling the forwarding is free of charge, but you will be charged for the usage of Logs Data Platform service as per standard price plan. For LDP pricing refer to [Logs Data Platform product page](https://www.ovhcloud.com/en-ie/logs-data-platform/).

> Note that the forwarding enablement is free of charge, but you will be charged for the usage of Logs Data Platform service as per standard price plan. For LDP pricing refer to [Logs Data Platform product page](https://www.ovhcloud.com/en-ie/logs-data-platform/).

Reviewer: ```suggestion
Note that enabling the forwarding is free of charge, but you will be charged for the usage of Logs Data Platform service as per standard price plan. For LDP pricing refer to [Logs Data Platform product page](https://www.ovhcloud.com/en-ie/logs-data-platform/).
```

To enable forwarding, you can use the following APIs:
|**Method**|**Log type**|**Path**|**Description**|
| :-: | :-: | :-: | :-: |
POST|Audit logs|/me/logs/audit/forward|Forward account audit logs|
POST|Activity logs|/me/api/logs/forward|Forward API & Customer panel account logs|
POST|Access policy logs|/iam/logs/forward|Forward account IAM logs to a dedicated logs stream|

For example for Audit logs :

```json
POST /me/logs/audit/forward
{
    "streamId": "ab51887e-0b98-4752-a514-f2513523a5cd"
}
```

> The API require a *streamId*, which is the target data stream of your LDP account where you aim your OVHcloud account logs will be forwarded. You will get in response an *operationid*, so you can use it to retrieve the *subscriptionid* further management purposes using [Logs Data Platform read operation endpoint](https://api.ovh.com/console-preview/?section=%2Fdbaas%2Flogs&branch=v1#get-/dbaas/logs/-serviceName-/operation). 

Reviewer: ```suggestion
The API requires a *streamId*. This is the target data stream of your LDP account where your OVHcloud account logs will be forwarded to. You will get in response an *operationid*, so you can use it to retrieve the *subscriptionid* further management purposes using [Logs Data Platform read operation endpoint](https://api.ovh.com/console-preview/?section=%2Fdbaas%2Flogs&branch=v1#get-/dbaas/logs/-serviceName-/operation). 
```
I think there's a word missing somewhere in the last sentence, I don't understand it

> You can find your streamId, in Logs Data Platform control panel of your account in Data stream” pan when editing a stream:

Reviewer: ```suggestion
You can find your streamId in the Logs Data Platform control panel :
```

- Go to the “Data Stream” page of your Logs Data Platform account and “Edit” the target Data Stream
![Find stream ID](images/retrieve_streamId_1.png){.thumbnail}

- Copy the *streamId* page of your Logs Data Platform account.
![Find stream ID](images/retrieve_streamId_2.png){.thumbnail}

Alternatively, you can retrieve your streamed using Logs Data Platform API:
> [!api]
>
> @api {GET} /dbaas/logs/{serviceName}/output/graylog/stream
> @api {GET} /dbaas/logs/{serviceName}/output/graylog/stream/{streamId}
>

> ### Access to OVHCLOUD Account logs

Reviewer: ```suggestion
### Access to OVHcloud Account logs
```
? 
I'm not sure about what the proper casing is for this, but it should probably be consistent throughout the document

> Now that your OVHCLOUD Account logs are ingested and stored in your Logs Data Platform data stream, you can query your logs and build dashboards to have a graphical representation of your logs using web-based UI of Graylog. 

Reviewer: ```suggestion
Now that your OVHcloud Account logs are ingested and stored in your Logs Data Platform data stream, you can query your logs and build dashboards to have a graphical representation of your logs using web-based UI of Graylog. 
```
This is a long sentence. Maybe you can rephrase it in 2 sentence to make it simpler ?

> -	Retrieve the admin user (<> Logs Data Platform service) and its password in your Logs Data Platform account home page.

Reviewer: what is "<>" supposed to mean here ? If it's as an alternative to "!=", I think we should avoid using coding symbols in the main text

Reviewer 2: Plz avoid using any mathematical/code format symbols in text (+->< etc.), not only for MD reasons but as a general best practice.

> -	Open the Graylog web-ui. You can retrieve the linke in your account home page or using your Access point depending of your account region (for example: Gravelines regions https://gra1.logs.ovh.com/ )

Reviewer: ```suggestion
-	Open the Graylog web-ui. You can retrieve the link from your account home page, or using your Access point depending of your account region (for example: Gravelines regions https://gra1.logs.ovh.com/ )
```

![Access Graylog](images/access_graylog.png){.thumbnail}

-	Login Graylog using your **Logs Data Platform service name** and **Password**

![Access Graylog](images/access_graylog_2.png){.thumbnail}

> -	Search your through your logs across the data stream of your Logs Data Platform account. You can refer to [Graylog writing search queries documentation](https://go2docs.graylog.org/4-x/making_sense_of_your_log_data/writing_search_queries.html?tocpath=Searching%20Your%20Log%20Data%7C_____1) for details on search syntax.

Reviewer: Missing word on the first sentence here

![Access Graylog](images/access_graylog_3.png){.thumbnail}

For more details about how to use your logs with Logs Data Platform, including:
-	How to setup alerts, 
-	View the logs in real time through a WebSocket, 
-	Build visualization with OpenSearch Dashboards, 
-	Integrate with OpenSearch API 
-	To connect with Grafana 

Refer to according documentations on `Manage and Operate` > `Observability` > `Logs Data Platform` > `Visualizing, querying and exploiting your logs` section.

### Details of logs generated

#### Audit logs

For every action related to the security of the OVHcloud account an entry is generated.

- **For login**

![Audit log](images/Audit_log_login.png){.thumbnail}

> Logs generated are:

Reviewer: This should be above L116 I think ?

|service_name|String|OVHcloud services concern by the action|
|source|manager or api|If the action were performed through the Control Panel (manager) or the API (api)|
|status_int|Number|HTTP code of request result|
|url|String|URL called on the action|
|user_agent|String|User agent of the user who performed the action|
|username|String|Username of the user who performed the action|

#### Access Policies logs

For every action evaluated by the OVHcloud IAM, an entry is generated with the following data:

![Activity log](images/Access_Policies_logs.png){.thumbnail}

|**Field**|**Value**|**Description**|
| :-: | :-: | :-: |
|account|String|OVHcloud account concern by the action
|identities_array|Array of String|List of identities of the user who performed the action (user URN and user’s groups URN)|
|requested_actions_array|Array of String|List of actions requested by the user|
|resource|String|URN of the OVHcloud resource concerned by the action|
|authorized_actions_array|Array of String|List of actions authorized after policy evaluation|
|unauthorized_actions_array|Array of String|List of actions unauthorized after policy evaluation|

### Manage subscriptions

> At any point, you can retrieve subscriptions attached to your Logs Data Platform data stream and choose to disable the forwarding by cancelling your subscription on your stream so that your Logs Data Platform stream doesn't receive your audit logs anymore. 

Reviewer: Quite a long sentence

---

> title: OVHcloud Account logs Forwarding to LDP

Reviewer 3: Not sure customers knows a lot what LDP means

Reviewer 3: More generally, the title describe the technical action. Not why the customer wan to do it.
Antoher proposition: "Analyse your IAM activity with Logs Data Platform"

Reviewer 2: Best practice: Write full term at the beginning (Objective): Logs Data Platform (LDP). Below in text you can use the acro. 
For frontmatter: Use full term at least once (title, excerpt).

excerpt: "Forwarding OVHcloud Account logs to Logs Data Platform"
updated: 2023-10-23
---

 

## Objective

In this guide, you will learn how to transform your OVHcloud Account events as actionable data through Logs Data Platform, a fully managed solution that helps you store, archieve, query and visualize your logs.  
To discover Logs Data Platform before continuing this guide, please refer to [this documentation](/pages/manage_and_operate/observability/logs_data_platform/getting_started_introduction_to_LDP).

> 

Reviewer 3: Maybe a small warning here: 

"This action will inject data into Logs Data Platform and could generate fees."

Author: Warning is at the begining of the "Enable logs forwarding" section

**Logs forwarding**: feature integrated in an OVHcloud product to ingest logs from its services to a data stream of an LDP account in the same OVHcloud account. The feature has to be enabled by the customer per service.

**Logs forwarding Subscription**: when enabling the logs forwarding for a given OVHcloud service to a given LDP data stream, a subscription is created and attached to the data stream for further management by the customer.

**OVH Account audit logs**: Log entries for sensible activities happening within your OVH Account customer main need to collect and use for compliance purposes and other business policy enforcement.

**OVH Account activity logs**: Log entries for all API calls done through OVHCloud API including those triggered when using OVHcloud manager.

**OVH Account policy logs**: Log entries for OVH services access acceptance or denied access due to of a security policy evaluation.

## Instruction

### OVHcloud Account logs type

> OVHcloud account propose 3 level of logs

Reviewer 3: levels

> - **Audit Logs**: provide a security-relevant, chronological set of records documenting the sequence of actions in your OVHcloud Account. (ie. logins, password change, …)

Reviewer 3: duplicate from glossary. Maybe you should just keep logs level definition only here.

Reviewer 3: Can we have the exhaustive list of actions that will be describe in theses logs?

- **Activity Logs**: provide all of records of actions in your OVHcloud Account. (ie. API calls and actions done on the control panel)

> - **Access policy logs**: provide all of records of access evaluation in your OVHcloud Account. (ie. Actions authorized or unauthorized through IAM policies)

Reviewer 3: From customer point of vue, when he should use activity vs access logs?

### Enable logs forwarding

> Currently, you can only enable the forwarding of the OVHcloud account logs via API. You will have to target a Stream of one of your LDP account. The logs will be forwarded to that Stream. Enabling the forwarding will create a subscription for this stream id.

Reviewer 3: Remove "currently"

Reviewer 3: LDP acronym is not defined below. You can use full name or define acronym. Personally, I prefer full name for people that don't read glossary ..

> Note that enabling the forwarding is free of charge, but you will be charged for the usage of Logs Data Platform service as per standard price plan. For LDP pricing refer to [Logs Data Platform product page](https://www.ovhcloud.com/en-ie/logs-data-platform/).

Reviewer 3: Seen.

Maybe this one should be from the beginning ?

To enable forwarding, you can use the following APIs:
|**Method**|**Log type**|**Path**|**Description**|
| :-: | :-: | :-: | :-: |
POST|Audit logs|/me/logs/audit/forward|Forward account audit logs|
POST|Activity logs|/me/api/logs/forward|Forward API & Customer panel account logs|
POST|Access policy logs|/iam/logs/forward|Forward account IAM logs to a dedicated logs stream|

> For example for Audit logs :

Reviewer 3: For instance,
